core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,36 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer,
+    private var task_ : Task[_] = null) extends  Logging {
```

Comment:
@squito
I agree with you, I will modify it.

@kayousterhout 
The main reason for this is to reduce the amount of code changes, we need to change more code files to implement what you say.